lib/application.js

```diff
@@ -300,22 +300,34 @@ app.engine = function(ext, fn){
  * See the Router#param() docs for more details.
  *
  * @param {String|Array} name
- * @param {Function} fn
+ * @param {Function|Array} fn
  * @return {app} for chaining
  * @api public
  */
 
 app.param = function(name, fn){
+  var args = slice.apply(arguments);
+  var name = args[0];
+  var fns = args.slice(1);
+
   this.lazyrouter();
 
   if (Array.isArray(name)) {
     name.forEach(function(key) {
-      this.param(key, fn);
+      var params = [key].concat(fns);
```

Comment:
`[key].concat(fns)` will inadvertinately allow for the arguments to be arrays, because it'll flatten in certain cases, i.e.:
```js
$ node -pe '[2].concat([1,2])'
[ 2, 1, 2 ]
```